astropy/utils/tests/test_misc.py

```diff
@@ -54,13 +54,24 @@ def test_skip_hidden():
 
 
 def test_JsonCustomEncoder():
+    from ... import units as u
     assert json.dumps(np.arange(3), cls=misc.JsonCustomEncoder) == '[0, 1, 2]'
     assert json.dumps(1+2j, cls=misc.JsonCustomEncoder) == '[1.0, 2.0]'
     assert json.dumps(set([1, 2, 1]), cls=misc.JsonCustomEncoder) == '[1, 2]'
     assert json.dumps(b'hello world \xc3\x85',
                       cls=misc.JsonCustomEncoder) == '"hello world \\u00c5"'
     assert json.dumps({1: 2},
                       cls=misc.JsonCustomEncoder) == '{"1": 2}'  # default
+    assert json.dumps({1: u.m}, cls=misc.JsonCustomEncoder) == '{"1": "m"}'
+    # Quantities
+    tmp = json.dumps({'a': 5*u.cm}, cls=misc.JsonCustomEncoder)
+    assert '"unit": "cm"' in tmp # Order of dict keys is unknown here
```

Comment:
You should be able to do these tests in a more transparent and complete way, something like:
```
out = json.loads(json.dumps({'a': 5 * u.cm}, cls=misc.JsonCustomEncoder))
assert out == {'a': {'unit': 'cm', 'value': 5.0}}
```
Even though the key order might vary, two `dict`s can be compared like shown.  So the test requirement is not that some particular text is emitted but rather that the emitted text represents the JSON encoding of a specific Python data structure.